addons/l10n_se/models/res_company.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+import re
+
+
+class ResCompany(models.Model):
+    _inherit = 'res.company'
+
+    org_number = fields.Char(compute='_compute_org_number')
+
+    @api.depends('vat')
+    def _compute_org_number(self):
+        for company in self:
+            org_number = re.sub(r'\D', '', company.vat)[:-2]
+            org_number = org_number[:6] + '-' + org_number[6:]
```

Comment:
sure this works in all cases?

Reply:
Yes, it will work for swedish VAT. However it might be good to check if it's a Swedish VAT or not. The format for Swedish VAT is SEnnnnnnnnnc01 n = number and c = checksum. The format for Swedish Organisation Number is nnnnnn-nnnc as you might know the VAT is based on Org. number.